sql/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/ClientE2ETestSuite.scala

```diff
@@ -228,11 +228,12 @@ class ClientE2ETestSuite
   }
 
   test("spark deep recursion") {
+    var recursionDepth = if (System.getProperty("os.arch") == "s390x") 400 else 500
```

Comment:
Which Java distribution did you used on this `os.arch==s390x`?

> ClientE2ETestSuite: Change the recursion depth from 500 to 400 for s390x only since the jvm uses more memory on that platform.

Specifically, I'm wondering 
- if you tested both Java 17 and 21 on IBM Java distribution.
- if you tested other Java distributions (Zulu, Temurin, Oracle)

Reply:
Yes, please test it without Java 21 too.

For the other stuff, you can revise the PR description by narrowing down the scope. Currently, for me, it claims too much. The code itself looks reasonable to me. :)

Reply:
@dongjoon-hyun, I tested with Temurin21 and the OpenJDK21 from the ubuntu 22.04 distro and got the same results as with the version 17 jdks.

I updated the PR Description for ClientE2ETestSuite. I hope it matches what you are looking for. Happy to change it some more if not.